lib/traject/extraction_tools.rb

```diff
@@ -87,6 +87,40 @@ def extract_ordered_fields(rec, fields)
     extra_fields
   end
 
+  # method for when you have an subfield order preference for extraction
+  # fields must be separated with a space between datafield tag number and list of subfields
+  def extract_fields_strict_subfield_order(rec, fields)
+    fields = fields.split(':')
+    build_arr = []
```

Comment:
consider using inject() where you want to accumulate things in an array. a.inject(initial) {|m,x| m << f(x)} does what you want (where m is the accumulator/memo), but doesn't require all the local vars.